test/test_irods_http_api.py

```diff
@@ -1825,6 +1825,332 @@ def test_non_parallel_writes_with_data_exceeding_internal_write_threshold(self):
+                        'fields': {
+                            'op': 'write',
+                            'parallel-write-handle': parallel_write_handle,
+                            'offset': e[0] * count,
+                            'stream-index': e[0]
+                        },
+                        'bytes': e[1] * count
+                    }))
+
+                for f in concurrent.futures.as_completed(futures):
+                    result = f.result()
+                    self.logger.debug(result)
+                    self.assertEqual(result['irods_response']['status_code'], 0)
+
+            # End the parallel write.
+            r = requests.post(self.url_endpoint, headers=headers, data={
+                'op': 'parallel_write_shutdown',
+                'parallel-write-handle': parallel_write_handle
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+            # Read the contents of the data object and show it contains exactly what we expect.
+            r = requests.get(self.url_endpoint, headers=headers, params={
+                'op': 'read',
+                'lpath': data_object,
+                'count': 30
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.content.decode('utf-8'), 'A' * 10 + 'B' * 10 + 'C' * 10)
+
+        finally:
+            # End the parallel write in case something failed.
+            # This avoids leaking of parallel-write resources in the HTTP API server.
+            if invoke_parallel_write_shutdown:
+                r = requests.post(self.url_endpoint, headers=headers, data={
+                    'op': 'parallel_write_shutdown',
+                    'parallel-write-handle': parallel_write_handle
+                })
+                self.logger.debug(r.content)
+
+            # Remove the data object.
+            r = requests.post(self.url_endpoint, headers=headers, data={
+                'op': 'remove',
+                'lpath': data_object,
+                'catalog-only': 0,
+                'no-trash': 1
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+    def test_parallel_writes_using_header_based_form_with_data_exceeding_internal_write_threshold(self):
+        # This test assumes the HTTP API is configured to use a value smaller than
+        # 96kb for "/irods_client/max_number_of_bytes_per_write_operation". This is
+        # important because values greater than or equal to 96kb will result in the
+        # background thread pool only being used one time. This test is designed to
+        # force the HTTP API to schedule multiple writes across the thread pool.
```

Comment:
Is it possible to check the configuration and `raise` an `EnvironmentError` if this is the not the case?

Reply:
Yes, that's possible if we want to make it a rule that the config is always local to the test.

Reply:
Okay. I feel like programmatically enforcing our test prerequisites could be helpful. Then again, if the configuration were to change and this test fails, the comment is here to explain why the test may have failed. Up to you.

Reply:
After some discussion, we've decided to assume that the tests are always run in the same environment as the HTTP API. That is, the test suite should always be able to read the config file of the HTTP API.

The new tests which this rule applies to will be updated to raise an `EnvironmentError` if the rule is violated.

Reply:
Hmm, right after I started working to update the test, I realized that the location of the config file can be anywhere on the computer.

I'm going to hold on enforcing the prereqs and handle it as part of #395.

Reply:
Left a comment in #395 which points to this conversation.